functions/go/set-image/README.md

```diff
@@ -0,0 +1,124 @@
+# set-image
+
+## Overview
+
+<!--mdtogo:Short-->
+
+The `set-image` function sets the image for all instances of a provided image
+`name` with the provided image `newName:newTag` or `newName@digest`. A common
+deployment pattern is to update images to the latest tags or, ideally, digests.
+Sometimes people want to change the image repo, also, such as for dev builds.
+
+<!--mdtogo-->
+
+You can learn more about images [here][image].
+
+<!--mdtogo:Long-->
+
+## Usage
+
+This function can be used with any KRM function orchestrators (e.g. kpt).
+
+For all resources which specify an image of a given `name`, the `set-image`
+function sets that image field to the specified `newName:newTag` or
+`newName@digest`.
+
+By default the function updates the standard [fields][commonimage] used to
+specify an image.
+
+This function can be used both declaratively and imperatively.
+
+### FunctionConfig
+
+There are 2 kinds of `functionConfig` supported by this function:
+
+- `ConfigMap`
+- A custom resource of kind `SetImage`
+
+To use a `ConfigMap` as the `functionConfig`, the desired image spec must be
+specified using the following fields:
+- `data.name`: Original name of image to replace. This field is required.
```

Comment:
We should point out that the image name should not contain tag or digest.

Reply:
Done